qcfractal/interface/collections/dataset.py

```diff
@@ -163,16 +161,35 @@ def download(self, local_path: Optional[Union[str, Path]] = None, verify: bool =
 
         self.set_view(local_path)
 
-    def _form_index(self) -> None:
-        self._entry_index = pd.DataFrame([[entry.name, entry.molecule_id] for entry in self.data.records],
-                                         columns=["name", "molecule_id"])
+    def _get_data_records_from_db(self):
+        self._check_client()
+        response = self.client.get_collection(self.__class__.__name__.lower(),
+                                              self.name,
+                                              full_return=False,
+                                              heavy=True)
+        self.data.__dict__["records"] = response.data.records
```

Comment:
Can we add a note that this only works because `get_collection` builds a validated Dataset object and so `response.data.records` is correctly coerced to a Pydantic object.